src/Nox.Types.EntityFramework/Types/AutoNumber/AutoNumberDatabaseConfigurator.cs

```diff
@@ -30,34 +30,17 @@ public class AutoNumberDatabaseConfigurator : INoxTypeDatabaseConfigurator
     /// <param name="property">The NoxSimpleTypeDefinition representing the property being configured.</param>
     /// <param name="entity">The Entity to which the property belongs.</param>
     /// <param name="isKey">A flag indicating whether the property is a key property.</param>
-    public void ConfigureEntityProperty(
+    public virtual void ConfigureEntityProperty(
         NoxCodeGenConventions noxSolutionCodeGeneratorState,
         IEntityBuilder builder,
         NoxSimpleTypeDefinition property,
         Entity entity,
         bool isKey)
     {
-        // If a normal attribute or key then it should be auto-incremented.
-        // Otherwise If it's a foreign key of entity id type or relationship it shouldn't be auto-incremented.
-        var shouldAutoincrement =
-            isKey ||
-            entity.Attributes.Any(x => x.Name.Equals(property.Name, StringComparison.OrdinalIgnoreCase) && x.Type == property.Type);
-
-        if (shouldAutoincrement)
-        {
-            builder
-                .Property(property.Name)
-                .IsRequired(property.IsRequired)
-                .HasConversion<AutoNumberConverter>()
-                .ValueGeneratedOnAdd();
-        }
-        else
-        {
-            builder
-                .Property(property.Name)
-                .IsRequired(property.IsRequired)
-                .HasConversion<AutoNumberConverter>();
-        }
+        builder
```

Comment:
Without "ValueGeneratedOnAdd" on the Autonumber property it would not work on the SqlLite?

Reply:
Since ValueGeneratedOnAdd moved to database spesific configurators and it is conditionaly used, it is in SqliteAutoNumberDatabaseConfigurator also.